### PR TITLE
Add tests and error checking for playing cards

### PR DESCRIPTION
**Pull request by dhillonJas**
More tests are added for playing card backend and some error checking is handled.

### CODE REVIEW

**Comment by mdangnguyen** [comment]
You should rebase the branch on `dev` and resolve the conflicts.

File: digital-deck-server/routes/player.js [hunk 1]
```diff
@@ -8,46 +8,68 @@ router.post('/playcard', async function (req, res) {
...
+  table.addCardTop(card);
+  res.status(200).send(session);
+
 });
 
 router.post('/drawcard', async function (req, res) {
-    let sessionId = Number(req.body.sessionId);
-    let playerId = Number(req.body.playerId);
-    let numOfCards = Number(req.body.numOfCards);
+  let sessionId = Number(req.body.sessionId);
```
Comment (mdangnguyen) [inline digital-deck-server/routes/player.js:49]: You don't want to commit this.  You're undoing the changes I made on my branch.

Comment (dhillonJas) [reply digital-deck-server/routes/player.js:49]: I didn't change anything in this. It is just line number change I think. I don't know why it shows it here

Comment (mdangnguyen) [reply digital-deck-server/routes/player.js:49]: It's the spacing. Each line has 2 less space at the beginning. I guess it doesn't really matter.

**Review by mdangnguyen — APPROVED** [review]

**Pull request merged**